src/inventory/mod.rs

```diff
@@ -134,31 +144,37 @@ pub struct InventoryContext {
     pub has_inv_open: bool,
     pub player_inventory: Arc<RwLock<PlayerInventory>>,
     pub base_inventory: Arc<RwLock<BaseInventory>>,
+    mouse_position: Option<(f64, f64)>,
+    conn: Arc<RwLock<Option<Conn>>>,
```

Comment:
We should probably use `ArcSwapOption<RwLock<Conn>>` everywhere we are using 
`Arc<RwLock<Option<Conn>>>` rn, although that doesn't have to be done in this pr

Reply:
I agree, using Option is pretty dirty, maybe we could encapsulate the 'connectedness' of the connection inside itself and we can do checks on it and handle interactions with it either internally or return a bool to indicate whether the action succeeded or failed so the caller can handle both cases as it is rn